test/system/src/test/java/io/pravega/test/system/MetadataScalabilityTest.java

```diff
@@ -138,40 +135,52 @@ public void scalability() {
-                        indexes.add(new AtomicInteger(1));
-                    }
-                    return Futures.loop(() -> indexes.stream().allMatch(x -> x.get() < SCALES_TO_PERFORM - 1), () -> {
-                        // randomly generate a stream cut. 
-                        // Note: From epoch 1 till epoch SCALES_TO_PERFORM each epoch is made up of 10k segments
-                        // and the range is statically partitioned evenly. 
-                        // So a random, correct streamcut would be choosing NUM_SEGMENTS disjoint segments from NUM_SEGMENTS random epochs. 
-                        Map<Segment, Long> map = new HashMap<>();
-                        for (int i = 0; i < NUM_SEGMENTS; i++) {
-                            AtomicInteger index = indexes.get(i);
-                            index.set(index.get() + rand.nextInt(SCALES_TO_PERFORM - index.get()));
-                            map.put(listOfEpochs.get(index.get()).get(i), 0L);
-                        }
-
-                        StreamCut cut = new StreamCutImpl(stream, map);
-                        log.info("truncating stream at {}", map);
-                        return controller.truncateStream(SCOPE, STREAM_NAME, cut).
-                                thenCompose(truncated -> {
-                                    log.info("stream truncated successfully at {}", cut);
-                                    assert truncated;
-                                    // we will just validate that a non empty value is returned. 
-                                    return controller.getSuccessors(cut)
-                                                     .thenAccept(successors -> {
-                                                         log.info("Successors for streamcut {} are {}", cut, successors);
-                                                     });
-                                });
-                    }, executorService);
-                }).join();
+        scaleFuture.join();
+        
+        return listOfEpochs;
+    }
+     
+     void truncation(ControllerImpl controller, List<List<Segment>> listOfEpochs) {
+         int numSegments = getStreamConfig().getScalingPolicy().getMinNumSegments();
+         int scalesToPerform = getScalesToPerform();
+         Stream stream = new StreamImpl(SCOPE, getStreamName());
+
+         // try SCALES_TO_PERFORM randomly generated stream cuts and truncate stream at those 
+         // stream cuts. 
+         List<AtomicInteger> indexes = new LinkedList<>();
+         Random rand = new Random();
+         for (int i = 0; i < numSegments; i++) {
+             indexes.add(new AtomicInteger(1));
+         }
+         Futures.loop(() -> indexes.stream().allMatch(x -> x.get() < scalesToPerform - 1), () -> {
+             // randomly generate a stream cut. 
+             // Note: From epoch 1 till epoch SCALES_TO_PERFORM each epoch is made up of numOfSegments segments
```

Comment:
Let's improve this description. Here is my recommendation:

```
We randomly generate a stream cut in each iteration of this loop. A valid stream
cut in this scenario contains for each position i in [0, numSegments -1], a segment
from one of the scale epochs of the stream. For each position i, we randomly
choose an epoch and pick the segment at position i. It increments the epoch
index accordingly (indexes list) so that in the next iteration it chooses a later
epoch for the same i.

Because the segment in position i always contain the range [d * (i-1), d * i],
where d = 1 / (number of segments), the stream cut is guaranteed to cover
the entire key space. 
```